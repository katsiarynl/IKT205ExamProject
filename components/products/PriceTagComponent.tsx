import { View, Text, StyleSheet } from "react-native";
import React from "react";
import products from "../../styles/products";
import { PriceTagType } from "../../types/productTypes";

export default function PriceTagComponent(props: PriceTagType) {
  return (
    <View style={styles.container}>
      <Text style={styles.price}>{props.price}</Text>
      <Text style={styles.currency}>NOK</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#FFA500", 
    borderRadius: 8,
    paddingHorizontal: 7,
    paddingVertical: 5,
    flexDirection: "row",
    alignItems: "center",
  },
  price: {
    color: "#FFFFFF", 
    fontSize: 16,
    fontWeight: "bold",
  },
  currency: {
    color: "#FFFFFF", 
    fontSize: 12,
    marginLeft: 5,
  },
});
