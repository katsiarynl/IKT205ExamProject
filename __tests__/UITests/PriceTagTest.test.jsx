import React from "react";
import { create } from "react-test-renderer";

import PriceTagComponent from "../../components/products/PriceTagComponent";
const tree = create(<PriceTagComponent price="55" />);

test("snapshot", () => {
  expect(tree).toMatchSnapshot();
});
