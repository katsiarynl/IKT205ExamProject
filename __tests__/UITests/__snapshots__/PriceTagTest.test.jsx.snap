// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFA500",
      "borderRadius": 8,
      "flexDirection": "row",
      "paddingHorizontal": 8,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    55
  </Text>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 12,
        "marginLeft": 5,
      }
    }
  >
    NOK
  </Text>
</View>
`;
